drivers/console/ipm_console.c

```diff
@@ -52,22 +53,30 @@ static void msgq_thread(void *arg1, void *arg2, void *arg3)
 
 static int console_out(int c)
 {
+	static char buf[CONFIG_IPM_CONSOLE_LINE_BUF_LEN];
+	static size_t len;
```

Comment:
Where is the locking for this data?  It something happening up the stack somewhere that makes this safe?  Remember that console output can happen from interrupt handlers and across SMP cores, and by default this code isn't safe.  I think you need a spinlock around this data.  If you don't need a spinlock, you need a comment explaining why.